### PR TITLE
fix: fixed integration tests

### PR DESCRIPTION
**Pull request by adubovik**
1. deprecated TEST_SERVER_URL env var
2. migrated to a fixture that provides test http client

### CODE REVIEW

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-bedrock-pr-147.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-bedrock-pr-147.nightly-test.deltixhub.io
>E2E tests status: success

File: pyproject.toml [hunk 1]
```diff
@@ -45,11 +45,15 @@ flake8 = "6.0.0"
...
 
 [tool.pytest.ini_options]
-addopts = "--doctest-modules"
+addopts="-n=auto"
+env_override_existing_values = 1
 # muting warnings coming from opentelemetry package
 filterwarnings = [
     "ignore::DeprecationWarning:opentelemetry.instrumentation.dependencies"
 ]
+testpaths = [
```
Comment (roman-romanov-o) [inline pyproject.toml]: What's the point of this option, if we explicitly state `"tests/unit_tests/` and `"tests/integration_tests/"` in noxfile?

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-bedrock-pr-147.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-bedrock-pr-147.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by roman-romanov-o — DISMISSED** [review]

File: pyproject.toml (hunk 1 above)
Comment (adubovik) [reply pyproject.toml]: fixed, removed

**Comment by adubovik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-adapter-bedrock-pr-147.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by roman-romanov-o — APPROVED** [review]

**Pull request merged**